pkg/database/db.go

```diff
@@ -83,61 +92,138 @@ func (o *Options) Validate() error {
-	expectedMysqlSchemaVersion    = 5
-	expectedPostgresSchemaVersion = 3
-)
+		config.DBName = c.Database
+		config.Timeout = time.Minute
+		config.Params = map[string]string{"sql_mode": "'TRADITIONAL,ANSI_QUOTES'"}
 
-// CheckSchema asserts the database schema of the expected version being present.
-func (db *DB) CheckSchema(ctx context.Context) error {
-	var expectedDbSchemaVersion uint16
-	switch db.DriverName() {
-	case MySQL:
-		expectedDbSchemaVersion = expectedMysqlSchemaVersion
-	case PostgreSQL:
-		expectedDbSchemaVersion = expectedPostgresSchemaVersion
-	}
+		tlsConfig, err := c.TlsOptions.MakeConfig(c.Host)
+		if err != nil {
+			return nil, err
+		}
 
-	var version uint16
+		config.TLS = tlsConfig
 
-	err := retry.WithBackoff(
-		ctx,
-		func(ctx context.Context) (err error) {
-			query := "SELECT version FROM icingadb_schema ORDER BY id DESC LIMIT 1"
-			err = db.QueryRowxContext(ctx, query).Scan(&version)
-			if err != nil {
-				err = internal.CantPerformQuery(err, query)
+		connector, err := mysql.NewConnector(config)
+		if err != nil {
+			return nil, errors.Wrap(err, "can't open mysql database")
+		}
+
+		onInitConn := connectorCallbacks.OnInitConn
+		connectorCallbacks.OnInitConn = func(ctx context.Context, conn driver.Conn) error {
+			if onInitConn != nil {
+				if err := onInitConn(ctx, conn); err != nil {
+					return err
+				}
+			}
+
+			return setGaleraOpts(ctx, conn, int64(c.Options.WsrepSyncWait))
+		}
+
+		addr = config.Addr
+		db = sqlx.NewDb(sql.OpenDB(NewConnector(connector, logger, connectorCallbacks)), MySQL)
+	case "pgsql":
+		uri := &url.URL{
+			Scheme: "postgres",
+			User:   url.UserPassword(c.User, c.Password),
+			Path:   "/" + url.PathEscape(c.Database),
+		}
+
+		query := url.Values{
+			"connect_timeout":   {"60"},
+			"binary_parameters": {"yes"},
+
+			// Host and port can alternatively be specified in the query string. lib/pq can't parse the connection URI
+			// if a Unix domain socket path is specified in the host part of the URI, therefore always use the query
+			// string. See also https://github.com/lib/pq/issues/796
+			"host": {c.Host},
+		}
+
+		port := c.Port
+		if !utils.IsUnixAddr(c.Host) {
+			if port == 0 {
+				port = 5432
 			}
-			return
-		},
-		retry.Retryable,
-		backoff.NewExponentialWithJitter(128*time.Millisecond, 1*time.Minute),
-		db.getDefaultRetrySettings())
-	if err != nil {
-		return errors.Wrap(err, "can't check database schema version")
-	}
 
-	if version != expectedDbSchemaVersion {
-		// Since these error messages are trivial and mostly caused by users, we don't need
-		// to print a stack trace here. However, since errors.Errorf() does this automatically,
-		// we need to use fmt instead.
-		return fmt.Errorf(
-			"unexpected database schema version: v%d (expected v%d), please make sure you have applied all database"+
-				" migrations after upgrading Icinga DB", version, expectedDbSchemaVersion,
-		)
+			query["port"] = []string{strconv.FormatInt(int64(c.Port), 10)}
```

Comment:
> You probably want to use `port` instead of `c.Port` here.

🤦

> Also, with PostgreSQL, the port setting is relevant even when using unix domain sockets.

Thanks, will be fixed.

> What you configure as the socket path everywhere is actually just the directory, the actual socket path is something like ...

So for PostgreSQL we will only log the directory.